sdk/storage/Azure.Storage.DataMovement.Blobs/src/PageBlobStorageResource.cs

```diff
@@ -317,14 +317,22 @@ protected override StorageResourceCheckpointDetails GetSourceCheckpointDetails()
         protected override StorageResourceCheckpointDetails GetDestinationCheckpointDetails()
         {
             return new BlobDestinationCheckpointDetails(
-                blobType: new(BlobType.Page),
+                isBlobTypeSet: true,
```

Comment:
I'm trying to think of a case where blob type would not be set... I feel like the destination checkpoint data should always save the Blob type and it shouldn't be "optional". Or how do we account for if the user does not want to preserve Blob type in the container options. Do container options need to trickle down to here?

Reply:
I had the same thought when I was changing blob type in particular. I need to find the line where if the blob type was set to default, where we assume it will be a block blob and maybe it will make less sense to have isblobtype on the checkpoint details since we always have to have a blob type, from either default or container options.

I thought the container options have already trickled down to this point, if we're at a blob item level.

Reply:
For container options
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobStorageResourceContainer.cs

in a lot of places we default to setting it to block blob if is not already set.

As for single transfers, the blob type needs to be specified when they are grabbing it from the provider (defaults to block as well).

Reply:
talked offline, we don't need to remove the isBlobTypeSet